source/solvers/tracer_assemblers.cc

```diff
@@ -345,3 +441,255 @@ TracerAssemblerBDF<dim>::assemble_rhs(TracerScratchData<dim>    &scratch_data,
+            {
+              copy_data_face.face_rhs(i) -=
+                fe_iv.jump_in_shape_values(i, q) // [\phi_i]
+                * scratch_data.values_here[q]    // \phi^{upwind}
+                * velocity_dot_n                 // (\beta .n)
+                * JxW[q];                        // dx
+            }
+          else
+            {
+              copy_data_face.face_rhs(i) -=
+                fe_iv.jump_in_shape_values(i, q) // [\phi_i]
+                * scratch_data.values_there[q]   // \phi^{upwind}
+                * velocity_dot_n                 // (\beta .n)
+                * JxW[q];                        // dx
+            }
+
+          // Assemble the diffusion term using Nitsche symmetric interior
+          // penalty method. See Larson Chap. 14. P.362
+          copy_data_face.face_rhs(i) -=
+            (scratch_data.tracer_diffusivity_face[q] *
+               (-scratch_data.tracer_average_gradient[q] * normals[q] *
+                  fe_iv.jump_in_shape_values(i, q) -
+                scratch_data.tracer_value_jump[q] * normals[q] *
+                  fe_iv.average_of_shape_gradients(i, q)) +
+             (scratch_data.tracer_diffusivity_face[q]) * penalty_factor *
+               scratch_data.tracer_value_jump[q] *
+               fe_iv.jump_in_shape_values(i, q)) *
+            JxW[q];
+        }
+    }
+}
+
+template class TracerAssemblerSIPG<2>;
+template class TracerAssemblerSIPG<3>;
+
+
+template <int dim>
+void
+TracerAssemblerBoundaryNitsche<dim>::assemble_matrix(
+  TracerScratchData<dim>      &scratch_data,
+  StabilizedDGMethodsCopyData &copy_data)
+{
+  const unsigned int           boundary_index = scratch_data.boundary_index;
+  const FEFaceValuesBase<dim> &fe_face =
+    scratch_data.fe_interface_values_tracer.get_fe_face_values(0);
+  const auto &q_points = fe_face.get_quadrature_points();
+
+  const double beta = scratch_data.penalty_factor;
+
+  const unsigned int         n_facet_dofs = fe_face.get_fe().n_dofs_per_cell();
+  const std::vector<double> &JxW          = fe_face.get_JxW_values();
+  const std::vector<Tensor<1, dim>> &normals = fe_face.get_normal_vectors();
+
+  for (unsigned int point = 0; point < q_points.size(); ++point)
+    {
+      const double velocity_dot_n =
+        scratch_data.face_velocity_values[point] * normals[point];
+      for (unsigned int i = 0; i < n_facet_dofs; ++i)
+        {
+          for (unsigned int j = 0; j < n_facet_dofs; ++j)
+            {
+              if (boundary_conditions_tracer.type[boundary_index] ==
+                  BoundaryConditions::BoundaryType::outlet)
+                {
+                  if (velocity_dot_n > 0)
+                    copy_data.local_matrix(i, j) +=
+                      fe_face.shape_value(i, point) *
+                      fe_face.shape_value(j, point) * velocity_dot_n *
+                      JxW[point];
+                }
+              if (boundary_conditions_tracer.type[boundary_index] ==
```

Comment:
```suggestion
             else if (boundary_conditions_tracer.type[boundary_index] ==
```

Reply:
No this is not an else if here, this is really an if :).

Reply:
Alright! A BC can be both outlet and dirichlet?

Reply:
Ahh no, you are right, it's either one or the other.